### ISSUE TITLE
RuntimeError: module compiled against API version a but this version of numpy is 9 E

### ISSUE THREAD

**Issue by switchfootsid** [cvxgrp/cvxpy#250]
Hi cvxgrp,
I just install cvxpy and tried running the nosetest and encountered the following errors. I've tried using several different versions of numpy, python, nose but haven't been able to get an error free test. Could you help me? I am using Python 2.7.10 and numpy 1.10.2.

RuntimeError: module compiled against API version a but this version of numpy is 9 E
======================================================================
ERROR: Failure: ImportError (numpy.core.multiarray failed to import)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/nose/loader.py", line 407, in loadTestsFromName
    module = resolve_name(addr.module)
  File "/Library/Python/2.7/site-packages/nose/util.py", line 312, in resolve_name
    module = __import__('.'.join(parts_copy))
  File "/Library/Python/2.7/site-packages/cvxpy/__init__.py", line 24, in <module>
    from cvxpy.problems.problem import Problem
  File "/Library/Python/2.7/site-packages/cvxpy/problems/problem.py", line 26, in <module>
    from cvxpy.problems.solvers.solver import Solver
  File "/Library/Python/2.7/site-packages/cvxpy/problems/solvers/solver.py", line 23, in <module>
    from cvxpy.problems.problem_data.matrix_data import MatrixData
  File "/Library/Python/2.7/site-packages/cvxpy/problems/problem_data/matrix_data.py", line 24, in <module>
    import canonInterface
  File "/Library/Python/2.7/site-packages/canonInterface.py", line 16, in <module>
    import CVXcanon
  File "/Library/Python/2.7/site-packages/CVXcanon.py", line 28, in <module>
    _CVXcanon = swig_import_helper()
  File "/Library/Python/2.7/site-packages/CVXcanon.py", line 24, in swig_import_helper
    _mod = imp.load_module('_CVXcanon', fp, pathname, description)
ImportError: numpy.core.multiarray failed to import

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)

**Comment by SteveDiamond** [comment]
Are you sure you're using numpy 1.10? Could you run ``pip list`` and paste the output here?

**Comment by switchfootsid** [comment]
Hi Steve, thanks for the speedy reply. I ran $pip list and got this output, seems like Numpy is fine:

altgraph (0.10.2)
appnope (0.1.0)
awscli (1.6.2)
backports-abc (0.4)
backports.ssl-match-hostname (3.5.0.1)
bcdoc (0.12.2)
bdist-mpkg (0.5.0)
bonjour-py (0.3)
boto (2.33.0)
botocore (0.73.0)
certifi (2015.11.20.1)
cffi (1.4.2)
charts (0.4.6)
colorama (0.2.5)
cryptography (1.1.2)
CVXcanon (0.0.23.3)
cvxopt (1.1.8)
cvxpy (0.3.4)
decorator (3.4.0)
docutils (0.12)
ecos (2.0.4)
enum34 (1.1.1)
functools32 (3.2.3-2)
gnureadline (6.3.3)
idna (2.0)
ipaddress (1.0.15)
ipykernel (4.2.1)
ipython (4.0.1)
ipython-genutils (0.1.0)
Jinja2 (2.8)
jmespath (0.5.0)
jsonschema (2.5.1)
jupyter-client (4.1.1)
jupyter-core (4.0.6)
librato-metrics (0.4.9)
macholib (1.5.1)
MarkupSafe (0.23)
matplotlib (1.3.1)
mistune (0.7.1)
mixpanel-py (3.1.1)
modulegraph (0.10.4)
multipledispatch (0.4.8)
nbconvert (4.1.0)
nbformat (4.0.1)
ndg-httpsclient (0.4.0)
nose (1.3.7)
notebook (4.0.6)
numpy (1.10.2)
pandas (0.17.1)
path.py (8.1.2)
pexpect (4.0.1)
pickleshare (0.5)
pip (7.1.2)
ply (3.8)
prettytable (0.7.2)
ptyprocess (0.5)
py2app (0.7.3)
pyasn1 (0.1.9)
pycparser (2.14)
Pygments (2.0.2)
pymc3 (3.0)
pyobjc-core (2.5.1)
pyobjc-framework-Accounts (2.5.1)
pyobjc-framework-AddressBook (2.5.1)
pyobjc-framework-AppleScriptKit (2.5.1)
pyobjc-framework-AppleScriptObjC (2.5.1)
pyobjc-framework-Automator (2.5.1)
pyobjc-framework-CFNetwork (2.5.1)
pyobjc-framework-Cocoa (2.5.1)
pyobjc-framework-Collaboration (2.5.1)
pyobjc-framework-CoreData (2.5.1)
pyobjc-framework-CoreLocation (2.5.1)
pyobjc-framework-CoreText (2.5.1)
pyobjc-framework-DictionaryServices (2.5.1)
pyobjc-framework-EventKit (2.5.1)
pyobjc-framework-ExceptionHandling (2.5.1)
pyobjc-framework-FSEvents (2.5.1)
pyobjc-framework-InputMethodKit (2.5.1)
pyobjc-framework-InstallerPlugins (2.5.1)
pyobjc-framework-InstantMessage (2.5.1)
pyobjc-framework-LatentSemanticMapping (2.5.1)
pyobjc-framework-LaunchServices (2.5.1)
pyobjc-framework-Message (2.5.1)
pyobjc-framework-OpenDirectory (2.5.1)
pyobjc-framework-PreferencePanes (2.5.1)
pyobjc-framework-PubSub (2.5.1)
pyobjc-framework-QTKit (2.5.1)
pyobjc-framework-Quartz (2.5.1)
pyobjc-framework-ScreenSaver (2.5.1)
pyobjc-framework-ScriptingBridge (2.5.1)
pyobjc-framework-SearchKit (2.5.1)
pyobjc-framework-ServiceManagement (2.5.1)
pyobjc-framework-Social (2.5.1)
pyobjc-framework-SyncServices (2.5.1)
pyobjc-framework-SystemConfiguration (2.5.1)
pyobjc-framework-WebKit (2.5.1)
Pyomo (4.2.10784)
pyOpenSSL (0.15.1)
pyparsing (2.0.1)
python-dateutil (2.4.2)
pytz (2013.7)
PyUtilib (5.2.3601)
pyzmq (15.1.0)
requests (2.9.1)
rsa (3.1.2)
scikit-learn (0.17)
scipy (0.16.1)
scs (1.1.7)
setuptools (19.1.1)
SFrame (1.6)
simplegeneric (0.8.1)
singledispatch (3.4.0.3)
six (1.10.0)
terminado (0.5)
Theano (0.7.0)
toolz (0.7.4)
tornado (4.1)
traitlets (4.0.0)
Twisted (13.2.0)
vboxapi (1.0)
xattr (0.6.4)
zope.interface (4.1.1)

**Comment by SteveDiamond** [comment]
What happens if you just run ``import CVXcanon``?

**Comment by switchfootsid** [comment]
>>> import CVXcanon
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named CVXcanon
>>> import cvxpy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named cvxpy
>>>

**Comment by SteveDiamond** [comment]
If you can't import cvxpy you didn't install it. I suspect ``pip list`` is showing the packages for a different installation of Python than the one running as the Python prompt. Either uninstall all versions of Python besides the system Python or use Anaconda.

**Comment by switchfootsid** [comment]
So you were right about there being multiple versions of Python, pip list was giving me the system packages. So then I was able to install it directly from the git repo.  
*conda install --channel https://conda.anaconda.org/cvxgrp cvxpy*  
But it seemed to have downgraded my numpy:
>>numpy:    1.10.2-py27_0      --> 1.8.2-py27_0

**However** it still threw an error when I tried to *import CVXcanon* in my notebook. So I installed that via the conda package utility (from mcg/cvxcanon ) and it seemed to then upgrade my numpy back to the latest 1.10!  I will still have to run some examples to unearth some dependency errors in my system though or maybe proceed by uninstalling other versions of python, besides the System and Anaconda.

But I am able to successfully import both the libraries into my notebook now, thanks! :)
